Handler/Cancel.hs

```diff
@@ -0,0 +1,22 @@
+module Handler.Cancel where
+
+import Import
+import Stripe
+
+postCancelR :: Handler Html
+postCancelR = do
+    plan <- runDB $ getBy404 $ UniquePlan freePlanId
+    Entity userId user <- requireAuth
+
+    result <- runStripe $ cancelSubscription $ userStripeId user
+
+    case result of
+        Just _ -> do
+            runDB $ update userId [UserPlan =. entityKey plan]
+
+            setMessage "Plan changed!"
+            redirect RootR
+
+        _ -> do
```

Comment:
if `result` is not a `Just _`, the only other value it could have is `Nothing` right? Should that be used here instead?